tests/test_config_reader.py

```diff
@@ -295,7 +297,7 @@ def test_construct_stacks_constructs_stack(
             sceptre_user_data={},
             hooks={},
             s3_details=sentinel.s3_details,
-            dependencies=["child/level", "top/level"],
+            dependencies=["top/level", "child/level"],
```

Comment:
This changed simply because the previous merge call for dependencies used the opposite order of inputs to the strategy function compared to the one done in `_recursive_read`.